test/framework/maven.go

```diff
@@ -0,0 +1,162 @@
+func CreateMavenCommand(directory string) MavenCommand {
+	return &mavenCommandStruct{directory: directory}
+}
+
+// MavenCommand wraps information about the maven command to execute.
+type MavenCommand interface {
+	// WithLoggerContext method attaches a logger context to trace all the command logs when executing it.
+	WithLoggerContext(loggerContext string) MavenCommand
+	// Execute command and returns the outputs.
+	Execute(targets ...string) (string, error)
+
+	// SkipTests will skip testing automatically
+	SkipTests() MavenCommand
+	// Profiles sets the profiles to execute
+	Profiles(profiles ...string) MavenCommand
+}
+
+type mavenCommandStruct struct {
+	directory     string
+	loggerContext string
+
+	profiles     []string
+	otherOptions []string
+	skipTests    bool
+}
+
+func (mvnCmd *mavenCommandStruct) WithLoggerContext(loggerContext string) MavenCommand {
+	mvnCmd.loggerContext = loggerContext
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) InDirectory(directory string) MavenCommand {
+	mvnCmd.directory = directory
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) SkipTests() MavenCommand {
+	mvnCmd.skipTests = true
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) Profiles(profiles ...string) MavenCommand {
+	mvnCmd.profiles = append(mvnCmd.profiles, profiles...)
+	return mvnCmd
+}
+
+func (mvnCmd *mavenCommandStruct) Execute(targets ...string) (string, error) {
+	var args []string
+
+	// Setup custom maven repository if defined
+	if len(config.GetCustomMavenRepoURL()) > 0 {
+		if err := mvnCmd.setMavenRepoURL(config.GetCustomMavenRepoURL()); err != nil {
+			return "", err
+		}
+	}
+
+	args = append(args, targets...)
+	if len(mvnCmd.profiles) > 0 {
+		args = append(args, fmt.Sprintf("-P%s", strings.Join(mvnCmd.profiles, ",")))
+	}
+	if len(mvnCmd.otherOptions) > 0 {
+		args = append(args, strings.Join(mvnCmd.otherOptions, " "))
+	}
+
+	cmd := CreateCommand(mavenCommandName, args...).InDirectory(mvnCmd.directory)
+
+	// Set logger context if exists
+	if len(mvnCmd.loggerContext) > 0 {
+		cmd.WithLoggerContext(mvnCmd.loggerContext)
+	}
+
+	return cmd.Execute()
+}
+
+func (mvnCmd *mavenCommandStruct) setMavenRepoURL(repoURL string) error {
+	// Create settings.xml in directory
+	if err := ioutil.WriteFile(fmt.Sprintf("%s/settings.xml", mvnCmd.directory), []byte(fmt.Sprintf(settingsFile, repoURL, repoURL)), 0644); err != nil {
+		return err
+	}
+
+	// Add option to get copied settings.xml file
+	mvnCmd.otherOptions = append(mvnCmd.otherOptions, "-ssettings.xml")
+
+	return nil
+}
+
+const (
+	repositoryXMLDescription = `<id>main-repository</id>
+								<name>main-repository</name>
+								<url>%s</url>
+								<layout>default</layout>
+								<releases>
+									<enabled>true</enabled>
+									<updatePolicy>always</updatePolicy>
+								</releases>
+								<snapshots>
+									<enabled>true</enabled>
+									<updatePolicy>always</updatePolicy>
+								</snapshots>`
+)
```

Comment:
Can we avoid to have so many white spaces at the beginning of each line? Maybe it would help to have these values in static files.

Reply:
will change that

Reply:
confirmed, I get a `expected statement, found '.'` syntax error

Reply:
updated, let me know what you think

I am not a big fan to have that in static files

Reply:
Looks better now, but can we maintain the indention?

Reply:
updated. let me know :)

Reply:
Perfect!